pgjdbc/src/main/java/org/postgresql/jdbc/BooleanTypeUtil.java

```diff
@@ -43,7 +68,7 @@ static boolean castToBoolean(final Object in) throws PSQLException {
       return fromString((String) in);
     }
     if (in instanceof Character) {
-      return fromCharacter((Character) in);
+      return fromString(in.toString());
```

Comment:
The existing behavior auto-unboxed for every comparison.
It certainly is possible that a switch statement or a bitset could be used. The question is a trade off between ease of maint vs best possible performance.